### PR TITLE
Annotate all InstructionTranslator

### PR DESCRIPTION
**Pull request by oulgen**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #131509

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

File: torch/_dynamo/variables/dicts.py [hunk 1]
```diff
@@ -5,7 +5,10 @@
 import functools
 import inspect
 import sys
-from typing import Dict, List, Optional
+from typing import Dict, List, Optional, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from torch._dynamo.symbolic_convert import InstructionTranslator
```
Comment (zou3519) [inline torch/_dynamo/variables/dicts.py:11]: why the `if TYPE_CHECKING`?

**Review by zou3519 — APPROVED** [review]

File: torch/_dynamo/variables/dicts.py (hunk 1 above)
Comment (oulgen) [reply torch/_dynamo/variables/dicts.py:11]: otherwise circular dependancy

File: torch/_dynamo/variables/builtin.py [hunk 2]
```diff
@@ -10,10 +10,13 @@
 import types
 from collections import defaultdict, OrderedDict
```
Comment (eellison) [inline torch/_dynamo/variables/builtin.py:11]: This still has `# mypy: ignore-errors` at the top of the file - intended ?

File: torch/_dynamo/variables/base.py [hunk 3]
```diff
@@ -2,7 +2,10 @@
 
 import collections
```
Comment (eellison) [inline torch/_dynamo/variables/base.py:3]: Same with this

**Review by eellison — COMMENTED** [review]
There's no removal of `# mypy: ignore-errors`

File: torch/_dynamo/variables/builtin.py (hunk 2 above)
Comment (oulgen) [reply torch/_dynamo/variables/builtin.py:11]: Yes because I only annotate tx on this PR, there's A TON more work to be done to clean these files

**Pull request closed without merging**